js_modules/dagit/packages/core/src/assets/AssetEvents.tsx

```diff
@@ -139,6 +180,17 @@ export const AssetEvents: React.FC<Props> = ({
     }
   }, [loadedPartitionKeys, materializations, observations, xAxis]);
 
+  const jobRunsThatDidntMaterializeAsset = jobRunsCounts.find(
+    (jrc) => jrc.stepKey === assetKey.path[assetKey.path.length - 1],
+  );
+
+  const runWhichFailedToMaterialize =
+    !jobRunsThatDidntMaterializeAsset &&
+    latestRuns.length &&
+    (!grouped.length || grouped[0].latest?.runId !== latestRuns[0].run?.id)
```

Comment:
I think there might be one additional check we need here. Technically I think `latestRuns[0]` has a stepKey property, and we want to make sure that we grab the latestRuns entry for the stepKey that corresponds to this asset.

The mapping from asset <> stepKey is a bit whacky though... I think `stepName === assetNode.opName` and we'd need to pass in the opName from the parent component where we have the definition loaded (along with repository)